kilkari/src/main/java/org/motechproject/nms/kilkari/domain/CallRetry.java

```diff
@@ -43,11 +43,20 @@
     @Field
     private SubscriptionOrigin subscriptionOrigin;
 
+    /**
+     * the time at which this callRetry was created (extracted from a RequestId)
+     * allows us to check for duplicate callRetry and end up with better CDR processing logic when dealing with
+     * bad data from IMI
+     */
+    @Field
+    private String timestamp;
```

Comment:
why not just store the whole request id instead of parsing it for a timestamp string?